README.md

```diff
@@ -388,92 +184,112 @@ In the below example, `logo.png` can be any static asset shipped with any npm pa
 
+In order for this option to work, `django-webpack-loader` requires the `request` object to be in the context, to be able to keep track of the generated chunks.
 
-`./webpack_production.config.js`
-```javascript
-var config = require('./webpack.config.js');
-var BundleTracker = require('webpack-bundle-tracker');
-
-config.output.path = require('path').resolve('./assets/dist');
+The `request` object is passed by default via the `django.template.context_processors.request` middleware with using the Django built-in templating system, and also with using Jinja2.
 
-config.plugins = [
-    new BundleTracker({filename: './webpack-stats-prod.json'})
-]
+If you don't have `request` in the context for some reason (e.g. using `Template.render` or `render_to_string` directly without passing the request), you'll get warnings on the console.
 
-// override any other settings here like using Uglify or other things that make sense for production environments.
+## Running the project
+We must run both back-end and front-end projects to fully utilize `django-webpack-loader` and `webpack-bundle-tracker`. For the Webpack pipeline, please refer to [this section](#compiling-the-front-end-assets).
 
-module.exports = config;
+Regarding the Django part, it should be run as a regular project.
+```bash
+python manage.py migrate
+python manage.py runserver
 ```
 
-`settings.py`
-```python
-if not DEBUG:
-    WEBPACK_LOADER.update({
-        'BUNDLE_DIR_NAME': 'dist/',
-        'STATS_FILE': os.path.join(BASE_DIR, 'webpack-stats-prod.json')
-    })
-```
+> You can also check [this example](https://github.com/django-webpack/django-webpack-loader/tree/master/examples/simple) on how to run a project with `django-webpack-loader` and `webpack-bundle-track`.
 
-<br><br>
+## Usage in production
+We recommend that you keep your local bundles and the stats file outside the version control, having a production pipeline that will compile the assets during the deployment phase. We also recommend Django's automatic collecstatic to be disabled during the deployment and that you manually run it when deploying via a script.
 
+However, production usage for this package is **fairly flexible**. Other approaches may include keeping the production bundles in the version control and take that responsibility from the automatic pipeline, however you must remember to always generate new bundles before pushing to remote. If you wish to follow this approach and also keep Django's collectstatic turned on, you can store the stats file and bundles in a directory that is added to the `STATICFILES_DIR`. This ensure that the generated bundles are automatically collected to the target directory.
 
-You can also simply generate the bundles on the server before running collectstatic if that works for you.
+## Code splitting
+In case you wish to use code-splitting, follow the recipe below on the Javascript side.
```

Comment:
Please link to an explanation of what is code splitting. Can be the webpack docs.